src/Sylius/Bundle/ResourceBundle/Controller/ResourcesCollectionProvider.php

```diff
@@ -48,13 +49,20 @@ public function get(RequestConfiguration $requestConfiguration, RepositoryInterf
     {
         $resources = $this->resourcesResolver->getResources($requestConfiguration, $repository);
 
-        if ($resources instanceof Pagerfanta) {
+        if ($resources instanceof ResourceGridView or $resources instanceof Pagerfanta) {
```

Comment:
yes seems to be good ! Moreover, we can ensure $resources is an instance of paginator below.